prophy/utils.py

```diff
@@ -24,19 +23,18 @@ def jsonize(struct_, ordered=True):
     try:
         if isinstance(struct_, prophy.composite.struct):
             if ordered:
-                return [(dsc[0], jsonize(getattr(struct_, dsc[0], None)), ordered) for dsc in struct_._descriptor]
+                return [(dsc.name, jsonize(getattr(struct_, dsc.name, None), ordered)) for dsc in struct_._descriptor]
             else:
-                return {dsc[0]: jsonize(getattr(struct_, dsc[0], None), ordered) for dsc in struct_._descriptor}
+                return {dsc.name: jsonize(getattr(struct_, dsc.name, None), ordered) for dsc in struct_._descriptor}
 
         elif isinstance(struct_, prophy.composite.union):
-            disctiminated = next((dsc[0] for dsc in struct_._descriptor if dsc[2] == struct_.discriminator), None)
-            assert disctiminated is not None, "Failed to get currently discriminated union field"
-            return jsonize(getattr(struct_, disctiminated, None), ordered)
+            discriminated = next((dsc.name for dsc in struct_._descriptor if dsc.discriminator == struct_.discriminator), None)
+            assert discriminated is not None, "Failed to get currently discriminated union field"
+            return jsonize(getattr(struct_, discriminated, None), ordered)
 
-        elif isinstance(struct_, six.string_types):
+        elif isinstance(struct_, (basestring, str)):
```

Comment:
here I have a question. 
Why do we have six module in prophy, and cannot use six package?
I have fixed it and defined string_types in prophy.six...

Reply:
Seems we have a problem. I cannot define string_types as defined in six package, because tox[check] gives me error of undefined 'basestring'. Looks like we need to use package...

Reply:
so do it exactly as in `prophyc/six.py`

Reply:
> Why do we have six module in prophy, and cannot use six package?

Afair due to fact that six module defines many more compatibility settings than prophy may ever need. There is `future` python project as well which eases the pain of writing in python6:
https://python-future.org/